packages/wallets/glow/src/adapter.ts

```diff
@@ -0,0 +1,270 @@
+
+    private _connecting: boolean;
+    private _wallet: GlowWallet | null;
+    private _publicKey: PublicKey | null;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' || typeof document === 'undefined'
+            ? WalletReadyState.Unsupported
+            : WalletReadyState.NotDetected;
+
+    constructor(_config: GlowWalletAdapterConfig = {}) {
+        super();
+        this._connecting = false;
+        this._wallet = null;
+        this._publicKey = null;
+        if (this._readyState !== WalletReadyState.Unsupported) {
+            const handler = (event: MessageEvent<any>) => {
+                if (typeof event.data === 'object' && event.data.__glow_loaded) {
+                    if (this._readyState !== WalletReadyState.Installed) {
+                        this._readyState = WalletReadyState.Installed;
+                        this.emit('readyStateChange', this._readyState);
+                    }
+                    window.removeEventListener('message', handler);
+                }
+            };
+
+            window.addEventListener('message', handler);
```

Comment:
Hmm. I like the approach, but I don't like that this will leave a dangling event listener. The adapters can be reinstantiated (this can easily happen when used with React), and this will lead to a memory leak.

However, I can see how the interval used by `scopePollingDetectionStrategy` could cause similar issues. 

https://github.com/solana-labs/wallet-adapter/issues/334 and https://github.com/solana-labs/wallet-adapter/issues/337 may be of some use.

Perhaps what we need is a teardown function for adapters, but maybe I'm not being imaginative enough. @steveluscher wdyt?

Reply:
Since I want to get this in soon and iterate on it, I'm happy to do whatever works for other wallets now. 

Should we just remove the event listener part?

Reply:
This will definitely leak memory as Jordan said, because the event handler holds a reference to `this`. No matter if the dApp wants to dispose of an old reference to `new GlowWalletAdapter`, it will stick around forever.

Like you said, Jordan, we'd have to enforce through `Base*WalletAdapter` that subclasses implement a destructor. While this would work, it will _definitely_ turn us all into janitors for years to come – we'll be finding places where folks forgot to call `destroy()`, or release a subscription, forever.

We'll probably have to end up doing something like this, though. I'm guessing we could minimize the damage by creating a ‘detection starter’ instance method on the base adapter that returns a dispose handle, then make all of our code call that and the dispose handle it returns when the time comes.

```
type DisposeHandle = () => void;
function startDetection(): DisposeHandle {
  function handler() { /* ... */ }
  window.addEventListener('...', handler);
  const diposePoller = scopePollingDetectionStrategy(/* ... */);
  let disposed = false;
  return () => {
    if (disposed) return;
    disposed = true;
    window.removeEventListener('...', handler);
    disposePoller();
  }
}
```

Anyway, we'll figure that out later @vpontis. I'd say, for now, just make sure to `removeEventListener` the `handler` where you `return true` from the `scopePollingDetectionStrategy`. Then, at least, the handler will die when Glow is detected.

Reply:
Updated, thanks Steve!